.github/workflows/release.yml

```
@@ -108,3 +114,45 @@ jobs:
         continue-on-error: true #continue for now until we see it working in action
         run: |
           gh release upload ${{ github.ref_name }} "build/output/elastic_apm_profiler_*.zip"
+
+  post-release:
+    needs: [ 'release-windows']
+    runs-on: ubuntu-latest
+    env:
+      GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      GIT_TAG: v${{ needs.release-windows.outputs.agent-version }}
+      NEW_BRANCH: update/${{ needs.release-windows.outputs.agent-version }}
+      TARGET_BRANCH: ${{ needs.release-windows.outputs.major-version }}.x
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          depth: 0
+
+      - name: Create GitHub Pull Request if minor release.
+        run: |
+          echo "as long as there is a major.x branch"
+          existed_in_local=$(git ls-remote --heads origin ${TARGET_BRANCH})
+          if [ -z ${existed_in_local} ]; then
+            echo -e "::warning::Target branch '${TARGET_BRANCH}' does not exist."
+            exit 0
+          fi
+          git checkout $GIT_TAG -b ${NEW_BRANCH}
```

I was thinking something like this might be nicer?

```suggestion
          git checkout ${TARGET_BRANCH} -b ${NEW_BRANCH}
          git format-patch -k --stdout ${TARGET_BRANCH}...main -- docs  | git am -3 -k
```

This will only apply changes from the `docs` directory to the backport to `M.x`. 

Several benefits would be:

- speeding up required checks on the PR. The code in the commit was already tested on main and as part of the release, and we never release from `M.x` 
- The PR will only go up if `docs` actually had any changes.


From: https://stackoverflow.com/questions/19821749/git-cherry-pick-or-merge-specific-directory-from-another-branch